classes/dl-srcpkg.bbclass

```diff
@@ -0,0 +1,57 @@
+#
+# This class enables getting source code of all packages which are included in rootfs.
+#
+# Copyright (C) Cybertrust Japan Co., Ltd.
+#
+# SPDX-License-Identifier: MIT
+#
+
+#
+# [How to use]
+# Add following line into the local.conf.
+# INHERIT += "dl-srcpkg"
+#
+
+prepare_local_isarapt() {
+    # Make a local copy of isar-apt repo
+    rm -rf "${ROOTFSDIR}/tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/*"
+    mkdir -p "${ROOTFSDIR}/tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/apt/${DISTRO}"
+    cp -Rf "${REPO_ISAR_DIR}/${DISTRO}/dists" "${ROOTFSDIR}/tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/apt/${DISTRO}/"
+    if [ -d "${REPO_ISAR_DIR}/${DISTRO}/pool" ]; then
+        cp -Rf "${REPO_ISAR_DIR}/${DISTRO}/pool" "${ROOTFSDIR}/tmp/isar-apt/${DISTRO}-${DISTRO_ARCH}/apt/${DISTRO}/"
+    fi
+}
+
+rootfs_generate_manifest:append () {
+    ### backup apt status
+    sudo sh -c "(cd '${ROOTFSDIR}' && tar zcpf apt_status.tar.gz var/cache/apt var/lib/apt)"
+
+    ### prepare directory for output
+    mkdir -p ${ROOTFS_MANIFEST_DEPLOY_DIR}/sources
+    sudo -E mount --bind /tmp '${ROOTFSDIR}'/tmp
+    mkdir -p '${ROOTFSDIR}'/tmp/deploy_dir
+    sudo -E mount --bind '${ROOTFS_MANIFEST_DEPLOY_DIR}' '${ROOTFSDIR}'/tmp/deploy_dir
+
+    ### setup network & apt environment
+    sudo cp /etc/resolv.conf '${ROOTFSDIR}'/etc/resolv.conf
+    prepare_local_isarapt
+    sudo sh -c "echo 'deb [trusted=yes] copy:///tmp/isar-apt/emlinux-bookworm-arm64/apt/emlinux-bookworm isar main' > '${ROOTFSDIR}'/etc/apt/sources.list.d/isar-apt.list"
```

Comment:
please fix hardcoded architecture name in line 38 and 39.

Reply:
I replaced some hardcoded strings and omitted some needless quote.

Reply:
thanks.